### PR TITLE
Clean up some FIXMEs.

### PR DESCRIPTION
**Pull request by luqmana**
Remove some assorted FIXMEs in the codebase.

### CODE REVIEW

File: src/librustc/middle/mem_categorization.rs [hunk 1]
```diff
@@ -947,9 +947,7 @@ impl<'a, 'gcx, 'tcx> MemCategorizationContext<'a, 'gcx, 'tcx> {
                 let ref_ty = self.overloaded_method_return_ty(method_ty);
                 base_cmt = self.cat_rvalue_node(elt.id(), elt.span(), ref_ty);
 
-                // FIXME(#20649) -- why are we using the `self_ty` as the element type...?
-                let self_ty = method_ty.fn_sig().input(0);
-                (self.tcx().no_late_bound_regions(&self_ty).unwrap(),
+                (ref_ty.builtin_deref(false, ty::NoPreference).unwrap().ty,
```
Comment (pnkfelix) [inline src/librustc/middle/mem_categorization.rs]: does this represent a change in analysis behavior?

Comment (luqmana) [inline src/librustc/middle/mem_categorization.rs]: Not that I could see and passed `make check`. Also @nikomatsakis mentioned on the issue that he didn't expect any effect.

**Comment by alexcrichton** [comment]
ping r? @pnkfelix

**Comment by alexcrichton** [comment]
@eddyb, thoughts here? (looks fairly trivial)

@luqmana want to rebase?

**Comment by eddyb** [comment]
r=me w/ rebase.

**Pull request merged**